examples/__tests__/react-hooks.js

```diff
@@ -0,0 +1,46 @@
+import {testHook, cleanup} from 'react-testing-library'
+
+import useCounter from '../react-hooks'
+
+afterEach(cleanup)
+
+test('accepts default initial values', () => {
+  let count
+  testHook(() => ({count} = useCounter()))
```

Comment:
It might be nice if `testHook` could return the return value of `useCounter` rather than needing to fish it out ourselves like this. Thoughts?

Reply:
🤔 

```jsx
function TestHook({callback}) {
  callback()
  return null
}

function testHook(callback) {
  let result
  let handleReturn = () => { result = callback() }
  render(<TestHook callback={handleReturn} />)
  return result
}
```

```js
// test.js
let {count, increment} = testHook(useCounter)
increment()
expect(count).toBe(1)
```

Reply:
@kentcdodds @donavon thoughts on that implementation?

Reply:
what @kentcdodds said. this will only return the initial value returned by the hook. The code below won't work as `count` will still be `0`.

```js
let {count, increment} = testHook(() => useCounter(0))
increment()
expect(count).toBe(1)
```

So while it will work if your tests only test the initial value, it will add way too much confusion IMO. We should be this as-is.

Reply:
Makes sense, thanks!